<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html><head></head><body>
































































<div class="container-fluid main-container">




<div>

<div class="btn-group pull-right float-right">
<span>Code</span> <span class="caret"></span>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a rel="noopener" href="#">Show All Code</a></li>
<li><a rel="noopener" href="#">Hide All Code</a></li>
<li class="divider"></li>
<li><a rel="noopener" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Daniela_Quijano_Homework_12</h1>

</div>



<div class="section level2">
<h2>MACS2 peaks and FRiP scores</h2>
<p>###Q1.1a How many peaks were called by MACS2 for each of the two Androgren Receptor ChIP-Seq samples in your MACS2 outputs from last week?</p>
<p>To count number of peaks, count the lines in the narrowpeak file. SRR7207011 has 1932 peaks SRR7207017 has 20931 peaks</p>



<pre class="bash"><code>[dq2033@log-2 ngs.week11]$ wc -l SRR7207011_MACS_peaks.narrowPeak
1932 SRR7207011_MACS_peaks.narrowPeak</code></pre>




<pre class="bash"><code>[dq2033@log-2 ngs.week11]$ wc -l SRR7207017_MACS_peaks.narrowPeak
20931 SRR7207017_MACS_peaks.narrowPeak</code></pre>



<p>###Q1.1b</p>
<p>The mean peak width for each sample is: SRR7207017 225 SRR7207011 333 The procedure used to find the mean peak width is below:</p>



<pre class="r"><code>#Open narroe peak file as a dataframe and reassign the column names based on documentation
SRR7207017_narrowPeak &lt;- as.data.frame(read.table(&quot;/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/SRR7207017_MACS_peaks.narrowPeak&quot;,header = FALSE, sep=&quot;\t&quot;,stringsAsFactors=FALSE, quote=&quot;&quot;))</code></pre>




<pre class="r"><code>column_fields&lt;-c(&#39;chrom&#39;,&#39;chromStart&#39;,&#39;chromEnd&#39;,&#39;name&#39;,&#39;score&#39;,&#39;strand&#39;,&#39;signalValue&#39;,&#39;pValue&#39;,&#39;qValue&#39;,&#39;peak&#39;)
colnames(SRR7207017_narrowPeak)&lt;-column_fields
head(SRR7207017_narrowPeak)</code></pre>


<div>

</div>




<pre class="r"><code>SRR7207011_narrowPeak &lt;- as.data.frame(read.table(&quot;/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/SRR7207011_MACS_peaks.narrowPeak&quot;,header = FALSE, sep=&quot;\t&quot;,stringsAsFactors=FALSE, quote=&quot;&quot;))
colnames(SRR7207011_narrowPeak)&lt;-column_fields
head(SRR7207011_narrowPeak)</code></pre>


<div>

</div>






<pre class="r"><code>#Define a variable/column made by chromEnd-chromStart
peak_width_SRR7207017&lt;-SRR7207017_narrowPeak$chromEnd-SRR7207017_narrowPeak$chromStart
sum(peak_width_SRR7207017)/length(peak_width_SRR7207017)</code></pre>


<pre><code>[1] 333.7633</code></pre>




<pre class="r"><code>peak_width_SRR7207011&lt;-SRR7207011_narrowPeak$chromEnd-SRR7207011_narrowPeak$chromStart
sum(peak_width_SRR7207011)/length(peak_width_SRR7207011)</code></pre>


<pre><code>[1] 225.7769</code></pre>



<p>###Q1.1c What is meant by the “signalValue” in column 7?</p>
<p>From ENCODE documentation, column 7, signal value is a ‘Measurement of overall (usually, average) enrichment for the region.’</p>
<p>###Q1.2 Report the FRiP score for each of the two androgen receptor ChIP-seq libraries and the command lines you used to generate your answer. Do the two androgen receptor libraries pass the 1% threshold typical of high quality ChIP-seq libraries? [ 2 points ].</p>
<p>-wo Write the original A and B entries plus the number of base pairs of overlap between the two features. Only A features with overlap are reported. Restricted by -f and -r.</p>



<pre class="bash"><code>[dq2033@log-2 ngs.week11]$ module load bedtools/intel/2.29.2 
[dq2033@log-2 ngs.week11]$ bedtools intersect -a SRR7207011_MACS_peaks.narrowPeak -b SRR7207011_sorted.bam -wo &gt; peaks_intersect.bed</code></pre>




<pre class="bash"><code>[dq2033@log-2 ngs.week11]$ head -3 peaks_intersect.bed 
1   2255338 2255695 SRR7207011_MACS_peak_1  96  .   7.22475 14.21419.65410  119 1   2255300 2255365 SRR7207011.14194    60  +   27
1   2255338 2255695 SRR7207011_MACS_peak_1  96  .   7.22475 14.21419.65410  119 1   2255306 2255371 SRR7207011.14195    60  +   33
1   2255338 2255695 SRR7207011_MACS_peak_1  96  .   7.22475 14.21419.65410  119 1   2255316 2255381 SRR7207011.14196    60  +   43</code></pre>




<pre class="bash"><code>[dq2033@log-1 ngs.week11]$ bedtools intersect -a SRR7207017_MACS_peaks.narrowPeak -b SRR7207089_sorted.bam -wo &gt; peaks_intersect_SRR7207017.bed</code></pre>




<pre class="bash"><code>[dq2033@log-1 ngs.week11]$ head -2 peaks_intersect_SRR7207017.bed
1   1433290 1433483 SRR7207017_MACS_peak_1  62  .   4.90366 9.252726.27751  141 1   1433345 1433410 SRR7207089.5959 60  -   65
1   1433290 1433483 SRR7207017_MACS_peak_1  62  .   4.90366 9.252726.27751  141 1   1433406 1433471 SRR7207089.5960 60  -   65</code></pre>



<p>Number intervals in the BAM file that intersect reads in the narrow peak file:</p>
<p>SRR7207011</p>



<pre class="bash"><code>[dq2033@log-2 ngs.week11]$ wc -l peaks_intersect.bed 
64445 peaks_intersect.bed</code></pre>



<p>SRR7207017</p>



<pre class="bash"><code>[dq2033@log-1 ngs.week11]$ wc -l peaks_intersect_SRR7207017.bed
89411 peaks_intersect_SRR7207017.bed</code></pre>



<p>FRIP= number of reads overlapping called peaks divided by the total number of reads.</p>



<pre class="bash"><code>[dq2033@log-1 ngs.week11]$ samtools flagstat SRR7207017_sorted.bam
21498262 + 0 in total (QC-passed reads + QC-failed reads)
21497944 + 0 primary
318 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
21497213 + 0 mapped (100.00% : N/A)
21496895 + 0 primary mapped (100.00% : N/A)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (N/A : N/A)
0 + 0 with itself and mate mapped
0 + 0 singletons (N/A : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ&gt;=5)</code></pre>






<pre class="bash"><code>[dq2033@log-1 ngs.week11]$ samtools flagstat SRR7207011_sorted.bam
23423985 + 0 in total (QC-passed reads + QC-failed reads)
23423635 + 0 primary
350 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
23422740 + 0 mapped (99.99% : N/A)
23422390 + 0 primary mapped (99.99% : N/A)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (N/A : N/A)
0 + 0 with itself and mate mapped
0 + 0 singletons (N/A : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ&gt;=5)</code></pre>



<p>I am using the total number of reads from each bam file obtained through samtools flagstat as seen above</p>
<p>Frip scores: SRR7207011-&gt; 64445/23423985 = 0.0027 SRR7207017-&gt; 89411/21498262 = 0.00415</p>
<p>Both of these scores do not pass the 1% Frip score guideline.</p>
<p>According to Landt et. al: The 1% FRiP guideline works well when there are thousands to tens of thousands of called occupancy sites in a large mammalian genome. However, passing this threshold does not automatically mean that an experiment is successful and a FRiP below the threshold does not automatically mean failure.</p>
</div>
<div class="section level2">
<h2>QC analysis of H3K36me3 ChIP-seq with the BioConductor ChIC package</h2>
<p>Download BAM files,copy files onto local directory</p>



<pre class="bash"><code>[dq2033@log-2 ngs.week11]$ cp /scratch/work/courses/BI7653/hw12.2022/ENCFF000BFX.bam .
[dq2033@log-2 ngs.week11]$ cp /scratch/work/courses/BI7653/hw12.2022/ENCFF000BDQ.bam .</code></pre>



<p>Download files from HPC to google drive</p>



<pre class="bash"><code>[dq2033@log-2 ngs.week11]$ rclone copy /scratch/dq2033/ngs.week11/ENCFF000BDQ.bam mygoogledrive:hpc_uploads
[dq2033@log-2 ngs.week11]$  rclone copy /scratch/dq2033/ngs.week11/ENCFF000BFX.bam mygoogledrive:hpc_uploads</code></pre>






<pre class="r"><code>```r
if (!require(\BiocManager\, quietly = TRUE))
    install.packages(\BiocManager\)

BiocManager::install(\ChIC\)</code></pre>
<pre><code>
&lt;!-- rnb-source-end --&gt;

&lt;!-- rnb-chunk-end --&gt;


&lt;!-- rnb-chunk-begin --&gt;


&lt;!-- rnb-source-begin eyJkYXRhIjoiYGBgclxubGlicmFyeShDaElDKVxuYGBgIn0= --&gt;

```r
library(ChIC)</code></pre>


<pre><code>Loading required package: spp
Loading required package: Rcpp
Registered S3 method overwritten by &#39;data.table&#39;:
  method           from
  print.data.table     </code></pre>


<pre class="r"><code>setwd(&quot;/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Week 12/chip_files&quot;)</code></pre>


<pre><code>Warning: The working directory was changed to /Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Week 12/chip_files inside a notebook chunk. The working directory will be reset when the chunk is finished running. Use the knitr root.dir option in the setup chunk to change the working directory for notebook chunks.</code></pre>






<pre class="r"><code>chipName &lt;- &quot;ENCFF000BFX&quot;
inputName &lt;- &quot;ENCFF000BDQ&quot;

chipBam &lt;- readBamFile(&quot;/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Week 12/chip_files/ENCFF000BFX&quot;)</code></pre>


<pre><code>9650808 reads from /Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Week 12/chip_files/ENCFF000BFX</code></pre>


<pre class="r"><code>inputBam &lt;- readBamFile(&quot;/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Week 12/chip_files/ENCFF000BDQ&quot;)</code></pre>


<pre><code>9983747 reads from /Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Week 12/chip_files/ENCFF000BDQ</code></pre>



<p>Define variables to run the analysis with multiple threads</p>



<pre class="r"><code>mc &lt;- 4
cluster &lt;- parallel::makeCluster( mc )</code></pre>



<p>Execute functions required for generating a cross-correlation profile plot. Cross-correlation profiling is a key method used by the ENCODE project to assess ChIP-seq library quality. The first steps to conducting cross-correlation profiling with ChIC is to determine binding peak separation distance and approximate window size that should be used for binding detection. Output variables are lists with information describing the cross-correlation profile. chip_binding.characteristics and input_binding.characteristics variables contain cross-correlation data required to construct the cross-correlation profile.</p>



<pre class="r"><code>chip_binding.characteristics &lt;- get.binding.characteristics(
    chipBam,
    srange=c(0,500),
    bin = 5,
    accept.all.tags = TRUE,
    cluster = cluster)
input_binding.characteristics &lt;- get.binding.characteristics(
  inputBam,
  srange=c(0,500),
  bin = 5,
  accept.all.tags = TRUE,
  cluster = cluster)
parallel::stopCluster( cluster )
</code></pre>



<p>Using the output binding characteristics from above, we can calculate the cross correlation QC-metrics (e.g., NSC and RSC) that are used to measure signal-to-noise for the ChIP sample and generate the cross-correlation plot. Execution of the command below should create a pdf in your working directory called CrossCorrelation.pdf.</p>



<pre class="r"><code>pdf(file = &quot;/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Homework/crosscorrelation_chipbam.pdf&quot;, width = 4, height = 4) 
crossvalues_Chip &lt;- getCrossCorrelationScores( chipBam ,
                                               chip_binding.characteristics,
                                               read_length = 36,
                                               annotationID=&quot;hg19&quot;,
                                               savePlotPath = &quot;.&quot;,
                                               mc = mc)</code></pre>


<pre><code>
hg19 valid annotation...
load chrom_info

(-) [=======&gt;-----------------------------------------]  17%
Phantom peak and cross-correlation...

(\) [===========&gt;-------------------------------------]  25%
(|) [===============&gt;---------------------------------]  33%
smoothing...
Check strandshift...
200
strandshift not adjusted...

(/) [===================&gt;-----------------------------]  42%
Phantom peak with smoothing...
Crosscorrelation plot saved under .
plot cross correlation curve with smoothing
pdf saved under./ChIPCrossCorrelation.pdf

(-) [=======================&gt;-------------------------]  50%
NRF calculation

(\) [============================&gt;--------------------]  58%
calculate alternative QC scores...

(|) [================================&gt;----------------]  67%
(/) [====================================&gt;------------]  75%
(-) [========================================&gt;--------]  83%
(\) [============================================&gt;----]  92%
(|) [=================================================] 100%</code></pre>


<pre class="r"><code>dev.off()</code></pre>


<pre><code>null device 
          1 </code></pre>




<pre class="r"><code>class(crossvalues_Chip)</code></pre>


<pre><code>[1] &quot;list&quot;</code></pre>


<pre class="r"><code>crossvalues_Chip</code></pre>


<pre><code>$CC_StrandShift
[1] 200

$tag.shift
[1] 100

$N1
[1] 8697346

$Nd
[1] 9035245

$CC_PBC
[1] 0.963

$CC_readLength
[1] 36

$CC_UNIQUE_TAGS_LibSizeadjusted
[1] 5988037

$CC_NSC
[1] 1.447

$CC_RSC
[1] 1.079

$CC_QualityFlag
[1] 1

$CC_shift
[1] 200

$CC_A
[1] 0.229

$CC_B
[1] 0.224

$CC_C
[1] 0.158

$CC_ALL_TAGS
[1] 9420858

$CC_UNIQUE_TAGS
[1] 9035245

$CC_UNIQUE_TAGS_nostrand
[1] 8909432

$CC_NRF
[1] 0.959

$CC_NRF_nostrand
[1] 0.946

$CC_NRF_LibSizeadjusted
[1] 0.599</code></pre>



<div class="section level3">
<h3>Q2.1b Select all that are true statements about the cross-correlation profile:</h3>
<p>aligned read asymmetries between DNA strands are the basis for the cross-correlation profile approach to ChIP-seq QC</p>
<p>correlations between strand-specific depths are recalculated after performing a “strand shift”</p>
<p>a single correlation is calculated between strand-specific depth</p>
<p>NSC values above a pre-defined threshold are of acceptable quality for downstream analysis</p>
</div>
<div class="section level3">
<h3>Q2.2a</h3>
<p>The NSC value for the chip salmple is 1.447</p>
</div>
<div class="section level3">
<h3>Q2.2b What is the RSC values for the ChIP sample?</h3>
<p>The RSC (Relative Strand Cross Correlation) values for this chip sample is 1.079</p>
</div>
<div class="section level3">
<h3>Q2.2c Do either of the metrics incorporate the “shaddow peak” height in how they are calculated? Which one(s)?</h3>
<p>The RSC value takes into consideration the ratio of the difference from the CHIP peak and the background and the phantom peak and the bakcground.</p>
</div>
<div class="section level3">
<h3>Q2.2d Landt et al.&#160;2012 (see “Cross-correlation Analysis”) provide minimum NSC and RSC values for libraries with acceptable signal-to-noise ratios. What are these minimum values and does this library pass ENCODE standards for quality control? Select one:</h3>
<p>minimum NSC = 1.05, minimum RSC = 0.8, the library passes QC</p>



<pre class="r"><code>crossvalues_input &lt;- getCrossCorrelationScores( inputBam,
                                               input_binding.characteristics,
                                               read_length = 36,
                                               annotationID=&quot;hg19&quot;,
                                               savePlotPath = &quot;.&quot;,
                                               mc = mc)</code></pre>


<pre><code>
hg19 valid annotation...
load chrom_info

(-) [=======&gt;-----------------------------------------]  17%
Phantom peak and cross-correlation...

(\) [===========&gt;-------------------------------------]  25%
(|) [===============&gt;---------------------------------]  33%
smoothing...
Check strandshift...
140
Strandshift is adjusted

(/) [===================&gt;-----------------------------]  42%
Phantom peak with smoothing...
Crosscorrelation plot saved under .
plot cross correlation curve with smoothing
pdf saved under./ChIPCrossCorrelation.pdf

(-) [=======================&gt;-------------------------]  50%
NRF calculation

(\) [============================&gt;--------------------]  58%
calculate alternative QC scores...

(|) [================================&gt;----------------]  67%
(/) [====================================&gt;------------]  75%
(-) [========================================&gt;--------]  83%
(\) [============================================&gt;----]  92%
(|) [=================================================] 100%</code></pre>




<pre class="r"><code>class(crossvalues_input)</code></pre>


<pre><code>[1] &quot;list&quot;</code></pre>


<pre class="r"><code>crossvalues_input</code></pre>


<pre><code>$CC_StrandShift
[1] 140

$tag.shift
[1] 70

$N1
[1] 8725518

$Nd
[1] 9119345

$CC_PBC
[1] 0.957

$CC_readLength
[1] 36

$CC_UNIQUE_TAGS_LibSizeadjusted
[1] 5992248

$CC_NSC
[1] 1.04

$CC_RSC
[1] 0.214

$CC_QualityFlag
[1] -2

$CC_shift
[1] 100

$CC_A
[1] 0.127

$CC_B
[1] 0.145

$CC_C
[1] 0.122

$CC_ALL_TAGS
[1] 9606580

$CC_UNIQUE_TAGS
[1] 9119345

$CC_UNIQUE_TAGS_nostrand
[1] 9098023

$CC_NRF
[1] 0.949

$CC_NRF_nostrand
[1] 0.947

$CC_NRF_LibSizeadjusted
[1] 0.599</code></pre>



</div>
<div class="section level3">
<h3>Q2.3a What steps are typically included/excluded in the preparation of the control/input sample? (see Week 11 pre-recorded video). Select all true statements.</h3>
<p>chromatin immunoprecipation with an appropriate antibody</p>
<p>With the input sample, a non specific pulldown is conducted sometimes with a mock IgG- Any IgG immunoprecipitated DNA should represent a random, nonspecific population.</p>
</div>
<div class="section level3">
<h3>Q2.3b NSC/RSC values for input sample</h3>
<p>1.04 NSC 0.214 RSC (due to low RSC value this would have failed QC)</p>
<div class="figure">
<img src="javascript://" alt=""/>
<p class="caption">Fingerprint Plot</p>
</div>
<div class="figure">
<img src="javascript://" alt=""/>
<p class="caption">Fingerprint Plot</p>
</div>
</div>
<div class="section level3">
<h3>Q2.4. ChIC calculates “Global Metrics” for ChIP-seq quality control.</h3>
<p>These metrics center around the fingerprint plot and nine metrics that can be derived from such plots</p>
<p>To generate a fingerprint plot, we need to calculate smoothed tag (read) density distributions. First remove anomalies in the BAMs:</p>



<pre class="r"><code>selectedTags &lt;- removeLocalTagAnomalies(chipBam,inputBam,chip_binding.characteristics)</code></pre>


<pre><code>Filter tags
SKIP select.informative.tags filter
remove.local.tag.anomalies filter</code></pre>




<pre class="r"><code>inputBamSelected &lt;- selectedTags$input.dataSelected
chipBamSelected &lt;- selectedTags$chip.dataSelected</code></pre>



<p>Extract the “tag shift” value from the cross-correlation output.</p>



<pre class="r"><code>finalTagShift &lt;- crossvalues_Chip$tag.shift
smoothedChip &lt;- tagDensity(chipBamSelected,
                           annotationID = &quot;hg19&quot;,
                           tag.shift = finalTagShift, mc = mc)</code></pre>


<pre><code>
hg19 valid annotation...
load chrom_info
load chrom_info
Smooth tag density</code></pre>


<pre class="r"><code>smoothedInput &lt;- tagDensity(inputBamSelected,
                            annotationID = &quot;hg19&quot;,
                           tag.shift = finalTagShift, mc = mc)</code></pre>


<pre><code>
hg19 valid annotation...
load chrom_info
load chrom_info
Smooth tag density</code></pre>



<p>Create a fingerprint plot and calculate global enrichment metrics</p>



<pre class="r"><code>Ch_Results &lt;- qualityScores_GM(selectedTagsChip = chipBamSelected,
                               selectedTagsInput = inputBamSelected,
                               savePlotPath = &quot;.&quot;,
                               tag.shift=finalTagShift,
                               annotationID=&#39;hg19&#39;)</code></pre>


<pre><code>***Calculating GM metrics...***

hg19 valid annotation...
load chrom_info
load chrom_info
Smooth tag density

hg19 valid annotation...
load chrom_info
load chrom_info
Smooth tag density
shorten frame...

(-) [===================&gt;-----------------------------]  40%
Calculate GM for ChIP ...

(\) [============================&gt;--------------------]  60%
Calculate GM for Input ...

(|) [======================================&gt;----------]  80%pdf saved under .

(/) [=================================================] 100%
Calculation of GM metrics done!</code></pre>



<p>The fingerprint plot produced:</p>
<p><img src="javascript://" alt="Fingerprint Plot"/> ### Fingerprint plot interpretation</p>
<p>the ChIP sample is higher quality because a greater proportion of reads are in higher ranked bins (i.e., bins ranked by depth) than the input NOTE: The chip library is also more complex than the input since it is farther away from the diagnoal. A uniform library would be mapped closer to the diagonal.</p>
<p>A common way of evaluating quality of ChIP-seq data and gaining biological insight into transcription factors or histone marks is to create a profile, or “enrichment” plot. A common type of plot in this category is the Transcription Start Site (TSS) profile plot which is described in the ChIC vignette categorized as a “Local Enrichment Metric (LM)”.</p>
<p>A TSS profile plot for the sample and control BAMs:</p>
<div class="figure">
<img src="javascript://" alt=""/>
<p class="caption">TSS Plot</p>
</div>
</div>
<div class="section level3">
<h3>Q2.5.</h3>
<p>What is shown in the TSS plot? Please provide a detailed interpretation of the plot. Based on the TSS profile plot, describe where are H3K36me3 modifications typically located relative to protein coding genes. Enrichment should increase to a single peak near TSS. In this case, the average depth around the transcription start site peaks at around 400 and 100 bases around the transcription start site-these locations correspond to the location of H3K6me3 modifications.</p>

</div>
</div>

<div>---
title: "Daniela_Quijano_Homework_12"
output: html_notebook
---

## MACS2 peaks and FRiP scores

###Q1.1a
How many peaks were called by MACS2 for each of the two Androgren Receptor ChIP-Seq samples in your MACS2 outputs from last week?

To count number of peaks, count the lines in the narrowpeak file.
SRR7207011 has 1932 peaks
SRR7207017 has 20931 peaks
```{bash}
[dq2033@log-2 ngs.week11]$ wc -l SRR7207011_MACS_peaks.narrowPeak
1932 SRR7207011_MACS_peaks.narrowPeak
```
```{bash}
[dq2033@log-2 ngs.week11]$ wc -l SRR7207017_MACS_peaks.narrowPeak
20931 SRR7207017_MACS_peaks.narrowPeak
```


###Q1.1b 


The mean peak width for each sample is:
SRR7207017 225
SRR7207011 333
The procedure used to find the mean peak width is below:

```{r}
#Open narroe peak file as a dataframe and reassign the column names based on documentation
SRR7207017_narrowPeak <- as.data.frame(read.table("/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/SRR7207017_MACS_peaks.narrowPeak",header = FALSE, sep="\t",stringsAsFactors=FALSE, quote=""))
```
```{r}
column_fields<-c('chrom','chromStart','chromEnd','name','score','strand','signalValue','pValue','qValue','peak')
colnames(SRR7207017_narrowPeak)<-column_fields
head(SRR7207017_narrowPeak)
```
```{r}
SRR7207011_narrowPeak <- as.data.frame(read.table("/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/SRR7207011_MACS_peaks.narrowPeak",header = FALSE, sep="\t",stringsAsFactors=FALSE, quote=""))
colnames(SRR7207011_narrowPeak)<-column_fields
head(SRR7207011_narrowPeak)
```

```{r}
#Define a variable/column made by chromEnd-chromStart
peak_width_SRR7207017<-SRR7207017_narrowPeak$chromEnd-SRR7207017_narrowPeak$chromStart
sum(peak_width_SRR7207017)/length(peak_width_SRR7207017)
```
```{r}
peak_width_SRR7207011<-SRR7207011_narrowPeak$chromEnd-SRR7207011_narrowPeak$chromStart
sum(peak_width_SRR7207011)/length(peak_width_SRR7207011)
```

###Q1.1c
What is meant by the “signalValue” in column 7?

From ENCODE documentation, column 7, signal value is a 'Measurement of overall (usually, average) enrichment for the region.'

###Q1.2
Report the FRiP score for each of the two androgen receptor ChIP-seq libraries and the command lines you used to generate your answer. Do the two androgen receptor libraries pass the 1% threshold typical of high quality ChIP-seq libraries? [ 2 points ].


-wo	Write the original A and B entries plus the number of base pairs of overlap between the two features. Only A features with overlap are reported. Restricted by -f and -r.


```{bash}
[dq2033@log-2 ngs.week11]$ module load bedtools/intel/2.29.2 
[dq2033@log-2 ngs.week11]$ bedtools intersect -a SRR7207011_MACS_peaks.narrowPeak -b SRR7207011_sorted.bam -wo > peaks_intersect.bed
```
```{bash}
[dq2033@log-2 ngs.week11]$ head -3 peaks_intersect.bed 
1	2255338	2255695	SRR7207011_MACS_peak_1	96	.	7.22475	14.21419.65410	119	1	2255300	2255365	SRR7207011.14194	60	+	27
1	2255338	2255695	SRR7207011_MACS_peak_1	96	.	7.22475	14.21419.65410	119	1	2255306	2255371	SRR7207011.14195	60	+	33
1	2255338	2255695	SRR7207011_MACS_peak_1	96	.	7.22475	14.21419.65410	119	1	2255316	2255381	SRR7207011.14196	60	+	43
```
```{bash}
[dq2033@log-1 ngs.week11]$ bedtools intersect -a SRR7207017_MACS_peaks.narrowPeak -b SRR7207089_sorted.bam -wo > peaks_intersect_SRR7207017.bed
```
```{bash}
[dq2033@log-1 ngs.week11]$ head -2 peaks_intersect_SRR7207017.bed
1	1433290	1433483	SRR7207017_MACS_peak_1	62	.	4.90366	9.252726.27751	141	1	1433345	1433410	SRR7207089.5959	60	-	65
1	1433290	1433483	SRR7207017_MACS_peak_1	62	.	4.90366	9.252726.27751	141	1	1433406	1433471	SRR7207089.5960	60	-	65
```



Number intervals in the BAM file that intersect reads in the narrow peak file:

SRR7207011
```{bash}
[dq2033@log-2 ngs.week11]$ wc -l peaks_intersect.bed 
64445 peaks_intersect.bed
```

SRR7207017
```{bash}
[dq2033@log-1 ngs.week11]$ wc -l peaks_intersect_SRR7207017.bed
89411 peaks_intersect_SRR7207017.bed
```

FRIP= number of reads overlapping called peaks divided by the total number of reads.
```{bash}
[dq2033@log-1 ngs.week11]$ samtools flagstat SRR7207017_sorted.bam
21498262 + 0 in total (QC-passed reads + QC-failed reads)
21497944 + 0 primary
318 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
21497213 + 0 mapped (100.00% : N/A)
21496895 + 0 primary mapped (100.00% : N/A)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (N/A : N/A)
0 + 0 with itself and mate mapped
0 + 0 singletons (N/A : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)
```

```{bash}
[dq2033@log-1 ngs.week11]$ samtools flagstat SRR7207011_sorted.bam
23423985 + 0 in total (QC-passed reads + QC-failed reads)
23423635 + 0 primary
350 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
23422740 + 0 mapped (99.99% : N/A)
23422390 + 0 primary mapped (99.99% : N/A)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (N/A : N/A)
0 + 0 with itself and mate mapped
0 + 0 singletons (N/A : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)
```

I am using the total number of reads from each bam file obtained through samtools flagstat as seen above 

Frip scores:
SRR7207011-> 64445/23423985 = 0.0027
SRR7207017-> 89411/21498262 = 0.00415

Both of these scores do not pass the 1% Frip score guideline. 

According to Landt et. al: The 1% FRiP guideline works well when there are thousands
to tens of thousands of called occupancy sites in a large mammalian
genome. However, passing this threshold does not automatically
mean that an experiment is successful and a FRiP below the threshold
does not automatically mean failure.

##  QC analysis of H3K36me3 ChIP-seq with the BioConductor ChIC package

Download BAM files,copy files onto local directory
```{bash}
[dq2033@log-2 ngs.week11]$ cp /scratch/work/courses/BI7653/hw12.2022/ENCFF000BFX.bam .
[dq2033@log-2 ngs.week11]$ cp /scratch/work/courses/BI7653/hw12.2022/ENCFF000BDQ.bam .
```
Download files from HPC to google drive
```{bash}
[dq2033@log-2 ngs.week11]$ rclone copy /scratch/dq2033/ngs.week11/ENCFF000BDQ.bam mygoogledrive:hpc_uploads
[dq2033@log-2 ngs.week11]$  rclone copy /scratch/dq2033/ngs.week11/ENCFF000BFX.bam mygoogledrive:hpc_uploads
```

```{r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("ChIC")
```
```{r}
library(ChIC)
setwd("/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Week 12/chip_files")
```

```{r}
chipName <- "ENCFF000BFX"
inputName <- "ENCFF000BDQ"

chipBam <- readBamFile("/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Week 12/chip_files/ENCFF000BFX")
inputBam <- readBamFile("/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Week 12/chip_files/ENCFF000BDQ")
```

Define variables to run the analysis with multiple threads
```{r}
mc <- 4
cluster <- parallel::makeCluster( mc )
```
Execute functions required for generating a cross-correlation profile plot. 
Cross-correlation profiling is a key method used by the ENCODE project to assess ChIP-seq library quality. 
The first steps to conducting cross-correlation profiling with ChIC is to determine binding peak separation distance and approximate window size that should be used for binding detection.
Output variables are lists with information describing the cross-correlation profile.
chip_binding.characteristics and input_binding.characteristics variables contain cross-correlation data required to construct the cross-correlation profile.
```{r}
chip_binding.characteristics <- get.binding.characteristics(
    chipBam,
    srange=c(0,500),
    bin = 5,
    accept.all.tags = TRUE,
    cluster = cluster)
input_binding.characteristics <- get.binding.characteristics(
  inputBam,
  srange=c(0,500),
  bin = 5,
  accept.all.tags = TRUE,
  cluster = cluster)
parallel::stopCluster( cluster )

```


Using the output binding characteristics from above, we can calculate the cross correlation QC-metrics (e.g., NSC and RSC) that are used to measure signal-to-noise for the ChIP sample and generate the cross-correlation plot. Execution of the command below should create a pdf in your working directory called CrossCorrelation.pdf.
```{r}
pdf(file = "/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Homework/crosscorrelation_chipbam.pdf", width = 4, height = 4) 
crossvalues_Chip <- getCrossCorrelationScores( chipBam ,
                                               chip_binding.characteristics,
                                               read_length = 36,
                                               annotationID="hg19",
                                               savePlotPath = ".",
                                               mc = mc)
dev.off()
```
```{r}
class(crossvalues_Chip)
crossvalues_Chip
```


### Q2.1b Select all that are true statements about the cross-correlation profile:

aligned read asymmetries between DNA strands are the basis for the cross-correlation profile approach to ChIP-seq QC

correlations between strand-specific depths are recalculated after performing a “strand shift”

a single correlation is calculated between strand-specific depth

NSC values above a pre-defined threshold are of acceptable quality for downstream analysis

### Q2.2a 
The NSC value for the chip salmple is 1.447

### Q2.2b What is the RSC values for the ChIP sample?
The RSC (Relative Strand Cross Correlation) values for this chip sample is  1.079

### Q2.2c Do either of the metrics incorporate the “shaddow peak” height in how they are calculated? Which one(s)?
The RSC value takes into consideration the ratio of the difference from the CHIP peak and the background and the phantom peak and the bakcground.

### Q2.2d Landt et al. 2012 (see “Cross-correlation Analysis”) provide minimum NSC and RSC values for libraries with acceptable signal-to-noise ratios. What are these minimum values and does this library pass ENCODE standards for quality control? Select one:

minimum NSC = 1.05, minimum RSC = 0.8, the library passes QC

```{r}
crossvalues_input <- getCrossCorrelationScores( inputBam,
                                               input_binding.characteristics,
                                               read_length = 36,
                                               annotationID="hg19",
                                               savePlotPath = ".",
                                               mc = mc)
```
```{r}
class(crossvalues_input)
crossvalues_input
```


### Q2.3a What steps are typically included/excluded in the preparation of the control/input sample? (see Week 11 pre-recorded video). Select all true statements.

chromatin immunoprecipation with an appropriate antibody

With the input sample, a non specific pulldown is conducted sometimes with a mock IgG- Any IgG immunoprecipitated DNA should represent a random, nonspecific population. 

### Q2.3b NSC/RSC values for input sample

1.04 NSC
0.214 RSC (due to low RSC value this would have failed QC)

![Fingerprint Plot](/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Plots_Images/ChIPCrossCorrelation.png)

![Fingerprint Plot](/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/Plots_Images/ChIPCrossCorrelation_input.png)

### Q2.4. ChIC calculates “Global Metrics” for ChIP-seq quality control.
These metrics center around the fingerprint plot and nine metrics that can be derived from such plots 

To generate a fingerprint plot, we need to calculate smoothed tag (read) density distributions. First remove anomalies in the BAMs:

```{r}
selectedTags <- removeLocalTagAnomalies(chipBam,inputBam,chip_binding.characteristics)

```
```{r}
inputBamSelected <- selectedTags$input.dataSelected
chipBamSelected <- selectedTags$chip.dataSelected
```

Extract the “tag shift” value from the cross-correlation output.

```{r}
finalTagShift <- crossvalues_Chip$tag.shift
smoothedChip <- tagDensity(chipBamSelected,
                           annotationID = "hg19",
                           tag.shift = finalTagShift, mc = mc)
smoothedInput <- tagDensity(inputBamSelected,
                            annotationID = "hg19",
                           tag.shift = finalTagShift, mc = mc)
```

Create a fingerprint plot and calculate global enrichment metrics

```{r}
Ch_Results <- qualityScores_GM(selectedTagsChip = chipBamSelected,
                               selectedTagsInput = inputBamSelected,
                               savePlotPath = ".",
                               tag.shift=finalTagShift,
                               annotationID='hg19')
```
The fingerprint plot produced:

![Fingerprint Plot](/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/FingerPrintPlot.png)
### Fingerprint plot interpretation

the ChIP sample is higher quality because a greater proportion of reads are in higher ranked bins (i.e., bins ranked by depth) than the input
NOTE: The chip library is also more complex than the input since it is farther away from the diagnoal. A uniform library would be mapped closer to the diagonal.


A common way of evaluating quality of ChIP-seq data and gaining biological insight into transcription factors or histone marks is to create a profile, or “enrichment” plot. A common type of plot in this category is the Transcription Start Site (TSS) profile plot which is described in the ChIC vignette categorized as a “Local Enrichment Metric (LM)”.

A TSS profile plot for the sample and control BAMs:

![TSS Plot](/Users/danielaquijano/Documents/NYU MASTERS/SPRING 2022/NGS Class/TSS_plot.png) 


### Q2.5.
What is shown in the TSS plot? Please provide a detailed interpretation of the plot. Based on the TSS profile plot, describe where are H3K36me3 modifications typically located relative to protein coding genes. 
Enrichment should increase to a single peak near TSS. In this case, the average depth around the transcription start site peaks at around 400 and 100 bases around the transcription start site-these locations correspond to the location of H3K6me3 modifications. 
</div>



</div>
















<script type="module" src="https://s.brightspace.com/lib/bsi/20.22.5-185/unbundled/mathjax.js"></script><script type="text/javascript">document.addEventListener('DOMContentLoaded', function() {
					if (document.querySelector('math') || /\$\$|\\\(|\\\[|\\begin{|\\ref{|\\eqref{/.test(document.body.innerHTML)) {
						document.querySelectorAll('mspace[linebreak="newline"]').forEach(elm => {
							elm.setAttribute('style', 'display: block; height: 0.5rem;');
						});

						window.D2L.MathJax.loadMathJax({
							'outputScale': 1.3,
							'renderLatex': true
						});
					}
				});</script></body></html>