# Next-Generation-Sequence-Analysis
The purpose of this repository is to store scripts, input and output files for NGS Analysis. Some of the analyses included in this repository are FASTQC analysis, genome alignment, SNP calling, and population genomics.

A short description of each of the folders within this repository:

## NGS Course Projects
This folder contains the projects completed in the course 'Next Generation Sequence Analysis' BIGY7653 at New York University. See below for a short description of each of the projects that have been completed and what the goals of each project were. 

### Population Genomics (W5)
-Assess genotype qualities in the Variant Call Format (VCF)
-Analyze likelihooods in low coverage sequencing analysis
-Calculate genotype likelihoods for population genomic analysis
-Estimate genetic ancestry in low coverage data

### Coverage Depth and Copy Number Variant Discovery (W6)
-Use Samtools stats program to determine coverage (average coverage genomewide) for paired-end (2 x 51 PE) reads from a Chlamydomonas strain (CC-2342)
-Calculate depth per base position using a single samtools depth command for the region chromosome_1:10001-10020 from the sample alignment for Chlamydomonas strain
-Calculate coverage depth in genomic intervals to identify a copy number variant 
